### DISCUSSION TITLE
How to use : beforeFormValidated in RelationManager - CreateAction ?

### DISCUSSION THREAD

**Discussion by jigujigu** [filamentphp/filament#6471] [Help]
Hi All,

I'm using RelationManager trying lifecycle hook.

I want to run my logic beforeFormValidated.
```
            ->headerActions([
                Tables\Actions\CreateAction::make()
                    ->beforeFormValidated(function (Model $record, array $data)  {
                        error_log('beforeFormValidated');
                       // dd ($data)    --> is empty result
                        dd($record)
                        // my logic for checking the $record or $data
                    })
            ])
```

But get: 
App\Filament\Resources\InvoiceResource\RelationManagers\DetailsRelationManager: 134
App\Filament\Resources\InvoiceResource\RelationManagers\{closure} : Argument #1 ($record) must be of type Illuminate\Database\Eloquent\Model, null given

Need advice, Thank you.

**Comment by danharrin** [comment]
The model has not been created at that point, so you cant use $record

**Reply by jigujigu** [reply]
Then, how or what variable I can use to get a value from input form  for my logic  ??

**Reply by danharrin** [reply]
Hmm, maybe `$form->getRawState()`, or after validation `$data` should work I guess

**Reply by bt760** [reply]
@danharrin  how can access to form variable in RelationManager class

**Reply by leandrocfe** [reply]
Where? What are you trying to do?
